vars/buildPlugin.groovy

```diff
@@ -123,23 +123,38 @@ def call(Map params = [:]) {
                   }
                   mavenOptions += 'clean install'
                   try {
+                    boolean noArtifactCachingProxyAvailable = true
                     if (artifactCachingProxyEnabled) {
                       // As azure VM agents don't have this env var, setting a default provider if none is specified or if the provider isn't available
                       final String defaultProxyProvider = 'azure'
-                      def availableProxyProviders = ['azure', 'do']
-                      String requestedProvider = env.ARTIFACT_CACHING_PROXY_PROVIDER
-                      if (requestedProvider == null || requestedProvider == '' || !availableProxyProviders.contains(requestedProvider)) {
-                        requestedProvider = defaultProxyProvider
-                        echo "INFO: no valid artifact caching proxy provider specified, set to '$defaultProxyProvider' by default."
+                      final String validProxyProviders = ['aws', 'azure', 'do']
+                      final String configuredAvailableProxyProviders = env.ARTIFACT_CACHING_PROXY_AVAILABLE_PROVIDERS
+                      String availableProxyProviders
+                      if (configuredAvailableProxyProviders != null && configuredAvailableProxyProviders != '') {
+                        availableProxyProviders = configuredAvailableProxyProviders.split(',')
+                        String requestedProvider = env.ARTIFACT_CACHING_PROXY_PROVIDER
+                        // Fallback to the default artifact caching proxy provider if the requested provider is empty, not available or not valid
+                        if (requestedProvider == null || requestedProvider == '' || !availableProxyProviders.contains(requestedProvider) || !validProxyProviders.contains(requestedProvider)) {
+                          echo "INFO: invalid or unavailable artifact caching proxy provider '${requestedProvider}' specified, set to '${defaultProxyProvider}' by default."
+                          requestedProvider = defaultProxyProvider
+                        } else {
+                          echo "INFO: using artifact caching proxy from '${requestedProvider}' provider."
+                        }
+
+                        if (availableProxyProviders.contains(requestedProvider)) {
+                          noArtifactCachingProxyAvailable = false
+                          configFileProvider(
+                              [configFile(fileId: "artifact-caching-proxy-${requestedProvider}", variable: 'MAVEN_SETTINGS')]) {
+                                infra.runMaven(mavenOptions, jdk, null, env.MAVEN_SETTINGS, addToolEnv)
+                              }
```

Comment:
- Since we introduce a fallback to repo.jenkins.io, the `defaultProxyProvider` does not make sense anymore. WDT about removing the "default" OR set it to `repo.jenkins-ci.org` (both choices would simplify the code path here)
- WDYT about the following simplified workflow:
  1. if env.ARTIFACT_CACHING_PROXY_PROVIDER is not empty AND in env.ARTIFACT_CACHING_PROXY_AVAILABLE_PROVIDERS.split(','), then test it. If test work then use it.
  2. otherwise use repo.jenkins-ci.org

Reply:
Having a defaultProvider set to azure is important as Azure VM agents can't be set with an env var.

Reply:
Can you add a comment then to explain why?

Otherwise WDYT of `requestProvider = env.ARTIFACT_CACHING_PROXY_PROVIDER ?: 'azure'` (groovy-isme -  https://docs.groovy-lang.org/latest/html/documentation/#_elvis_operator)

Reply:
note for later: we could set the value on Azure VM through the "Initialization Script", along with @smerle33 's datadog setup, by writing the value to `/home/jenkins/.profile` (or later `.bashrc` or whatever is eavluated in the context of a pipeline). WDYT @timja ?

Reply:
> Can you add a comment then to explain why?

There is already one L128 🙂 

> WDYT of requestProvider = env.ARTIFACT_CACHING_PROXY_PROVIDER ?: 'azure'

Looks nice!

Reply:
it should be do-able through the init script

Reply:
> even just detecting the vm is in Azure

Oh, do we have this feature in pipeline lanugage?